### DISCUSSION TITLE
Installation of Pmetrics on Linux (University Server)

### DISCUSSION THREAD

**Discussion by CDarlow** [LAPKB/Pmetrics#212] [Q&A]
Hi,

I'm trying to set up Pmetrics on our University High Performance Computer server, mostly to enable computational heavy model fitting with Pmetrics. The setup is a user space on a Linux system, with some pre-installed programmes (including R version 4.2.2. and GCC 11.2.0 compiler, but obviously not including the RStudio shell).

Does anyone have experience of installing Pmetrics in this sort of setting? I've managed to install the Pmetrics package by installing using the devtools install_github command. However, I'm having trouble compiling with PMBuild(). The command finds the locally installed gfortran compiler, but the compiling crashes out with the following:

> > PMbuild()
> Pmetrics cannot find required compiled binary files.
> /usr/bin/gfortran
> Pmetrics has detected gfortran and will compile required binary files.
> If the registration time outs, please re-run PMbuild(skipRegistration = TRUE) Pmetrics has anonymously registered your installation of this version.
> LAPKB does not collect or store any personal or identifying information.
> 
> Compiling 1 of 9: NPprep...
> Compiling 2 of 9: NPeng...NPeng_120.f:14828.9:
> 
>       use ieee_arithmetic !CR                                           
>          1
> Fatal Error: Can't open module file 'ieee_arithmetic.mod' for reading at (1): No such file or directory
> Error in PMbuild() : 
> There was an error compiling NPeng.

I don't know why the compiler isn't finding ieee_arithmetic.mod, which is in the gfortran compiler. Any help or insight would be appreciated.

**Comment by Siel** [comment]
Hi @CDarlow 
I actually use Pmetrics and develop for it under Linux, it should not be a problem to achieve what you're trying to do.
What I can see from your error message, it seems that you system might have an outdated version of gfortran, have you tried updating it? or maybe downloading a different one? Pmetrics by default will use the one in your PATH env variable, but you could specify a different one via `setPMoptions`

**Reply by CDarlow** [reply]
Thank you - I am using a managed remote server. The gfortran is uptodate (11.2), but I suspect it may be an issue with the wrong PATH. I will update when I've tried it, although I've managed to solve the current computational issue without needing to use the Linux server.
